File: abc133/abc133_c.cs
// https://atcoder.jp/contests/abc133/submissions/8264012

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var input = Console.ReadLine().Split(' ').Select(long.Parse).ToArray();
            var L = input[0];
            var R = input[1];

            var min = 2019L;
            var end = false;
            for (var i = L; i < R; i++)
            {
                for (var j = i + 1; j <= R; j++)
                {
                    if (i * j % 2019 < min)
                    {
                        min = i * j % 2019;
                        if (min == 0)
                        {
                            end = true;
                            break;
                        }
                    }
                }
                if (end)
                {
                    break;
                }
            }
            Console.WriteLine(min);
        }
    }
}

File: abc/abc133/abc133_c.fs
// https://atcoder.jp/contests/abc133/submissions/11968764

let [| L; R |] = stdin.ReadLine().Split() |> Array.map int64

let rec solver m i j =
    if i = R then m
    elif m = 0L then m
    elif j = R + 1L then solver m (i + 1L) (i + 2L)
    elif i * j % 2019L < m then solver (i * j % 2019L) i (j + 1L)
    else solver m i (j + 1L)

solver 2019L L (L + 1L) |> stdout.WriteLine
